sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2706,6 +2707,18 @@ class Dataset[T] private[sql](
 
   /**
    * Persist this Dataset with the default storage level (`MEMORY_AND_DISK`).
+   * @param eager If true, persist the Dataset eagerly.
+   * @group basic
+   * @since 2.3.0
+   */
+  def persist(eager: Boolean): this.type = {
```

Comment:
This varies from the RDD API then, which lets you control the persistence level, but doesn't expose this argument.